### PR TITLE
Assignment 2 Submission

### PR DESCRIPTION
**Pull request by cliodhnaharrison**
Assumptions:
1. No duplicate keys as stated in Slack.
2. Not a binary search tree.
3. Not a balanced tree.
4. Keys are in the tree.
5. Nodes are not ancestors of themselves based on example given in assignment.

### CODE REVIEW

File: cliodhnaharrison/assignment2/print_ancestors.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    if root == None:
+        return False
+
+    if root.value == key:
+        return True
+
+    if (print_ancestors(root.left, key) or print_ancestors(root.right, key)):
+        
+        #Prints ancestors when found
+        print (root.value)
```
[the remark this answers is not in the document]
Comment (cliodhnaharrison) [reply cliodhnaharrison/assignment2/print_ancestors.py]: Done, I now pass an empty list as a parameter and fill it with the ancestors recursively and return the list

File: cliodhnaharrison/assignment2/common_ancestors.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#Using Python 3
```
[the remark this answers is not in the document]
Comment (cliodhnaharrison) [reply cliodhnaharrison/assignment2/common_ancestors.py]: Have deleted that file and am working on adding the common ancestors function into the same file as print ancestors.

File: cliodhnaharrison/assignment2/ancestors.py [hunk 3]
```diff
@@ -15,6 +15,9 @@ def __init__(self, value):
 def print_ancestors(root, key, ancestors=[]):
     """
     Recursive function to print all ancestors of a Node.
+    Initially passed a tree, a key of a node and an empty list.
```
Comment (ferreron) [inline cliodhnaharrison/assignment2/ancestors.py]: You can follow the style guide and have a list of arguments, what the method returns, etc.

File: cliodhnaharrison/assignment2/ancestors.py [hunk 4]
```diff
@@ -24,15 +27,17 @@ def print_ancestors(root, key, ancestors=[]):
...
 
     if (print_ancestors(root.left, key, ancestors) or print_ancestors(root.right, key, ancestors)):
-        #Prints ancestors when found
+        #Adds ancestors to list when found
         ancestors.append(root.value)
         return ancestors
 
     return False
 
 def lowest_common_ancestors(root, node1, node2):
```
Comment (ferreron) [inline cliodhnaharrison/assignment2/ancestors.py]: Optional. If you want your code to work also when the nodes are not stored in the tree, you can simply add a method that given a value returns a node and use it as a first step before calling lowest_common_ancestors.

Comment (ferreron) [inline cliodhnaharrison/assignment2/ancestors.py]: To be more precise, node1 and node2 are values.

**Pull request merged**